### DISCUSSION TITLE
DRBG available error???

### DISCUSSION THREAD

**Discussion by myphj01** [openssl/openssl#25759] [Q&A] [answered]
Hello, I installed openssl (3.5), liboqs, and oqs-provider.
And after that, I tried to execute for generating keys.
But DBRG available issues comes out.

I tried twice and the following is captured at the second time.
At first, I tried with openssl 3.2 version.

--------------------------------------------------------
$ uname -a
Linux 6.8.0-47-generic #47-Ubuntu SMP PREEMPT_DYNAMIC Fri Sep 27 21:40:26 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/issue
Ubuntu 24.04.1 LTS \n \l
$ ls  ~/.local/lib64/ossl-modules/
legacy.so  oqsprovider.so
$ openssl version
OpenSSL 3.5.0-dev  (Library: OpenSSL 3.5.0-dev )
$ openssl req -x509 -new -newkey dilithium3 -keyout qsc.key -out qsc.crt -nodes -subj "/CN=oqstest" -days 365 -config ./openssl.cnf           
Warning: generating random key material may take a long time
if the system has a poor entropy source
No OpenSSL randomness retrieved. DRBG available?
$ cat openssl.cnf
openssl_conf = openssl_init

[openssl_init]
providers = provider_sect

[provider_sect]
oqsprovider = oqsprovider_sect

[oqsprovider_sect]
activate = 1

$ openssl rand -hex 16
c3538edf2f41fe9db45099555d9fcba9
$ openssl list -signature-algorithms -provider oqsprovider
  dilithium2 @ oqsprovider
  p256_dilithium2 @ oqsprovider
  rsa3072_dilithium2 @ oqsprovider
  dilithium3 @ oqsprovider
  p384_dilithium3 @ oqsprovider
  ...
  ...
$ openssl genpkey -algorithm mldsa87 -provider oqsprovider
Warning: generating random key material may take a long time
if the system has a poor entropy source
No OpenSSL randomness retrieved. DRBG available?
  

-------------------------------------------------------------

**Comment by mattcaswell** [comment, marked as answer]
The DRBG uses algorithms from the default provider. If you explicitly load another provider (in this case oqs provider), then you must also explicitly load the default provider if you plan to use it,. e.g.:

````
$ openssl genpkey -algorithm mldsa87 -provider oqsprovider -provider default
````
Update your config file similarly to additionally load the default provider.
